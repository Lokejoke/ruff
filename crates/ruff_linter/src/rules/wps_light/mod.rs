//! Rules from [wps-light](https://pypi.org/project/wps-light/).
pub(crate) mod rules;

#[cfg(test)]
mod tests {
    use std::convert::AsRef;
    use std::path::Path;

    use anyhow::Result;
    use test_case::test_case;

    use crate::registry::Rule;
    use crate::test::test_path;
    use crate::{assert_messages, settings};

    #[test_case(Rule::ListMultiplication, Path::new("WPS435.py"))]
    #[test_case(Rule::ReservedArgumentAsVariable, Path::new("WPS117.py"))]
    #[test_case(Rule::UnderscoresInNumber, Path::new("WPS303.py"))]
    #[test_case(Rule::AssignmentToSubscriptSlice, Path::new("WPS362.py"))]
    fn rules(rule_code: Rule, path: &Path) -> Result<()> {
        let snapshot = format!("{}_{}", rule_code.as_ref(), path.to_string_lossy());
        let diagnostics = test_path(
            Path::new("wps_light").join(path).as_path(),
            &settings::LinterSettings::for_rule(rule_code),
        )?;
        assert_messages!(snapshot, diagnostics);
        Ok(())
    }
}
