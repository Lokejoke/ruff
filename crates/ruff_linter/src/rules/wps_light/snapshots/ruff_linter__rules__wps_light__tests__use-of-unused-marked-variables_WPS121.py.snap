---
source: crates/ruff_linter/src/rules/wps_light/mod.rs
---
WPS121.py:52:5: WPS121 Local variable `_invalid_var` is marked as unused but is used
   |
51 | def fun(_valid_arg):
52 |     _invalid_var = _valid_arg  # [WPS121]
   |     ^^^^^^^^^^^^ WPS121
53 |     return _invalid_var
   |
   = help: Remove leading underscores from variable `_invalid_var`

WPS121.py:59:9: WPS121 Local variable `__` is marked as unused but is used
   |
57 |     def __init__(self):
58 |         _ = 1
59 |         __ = 2  # [WPS121]
   |         ^^ WPS121
60 |         ___ = 3
61 |         _invalid_type = int  # [WPS121]
   |
   = help: Remove leading underscores from variable `__`

WPS121.py:61:9: WPS121 Local variable `_invalid_type` is marked as unused but is used
   |
59 |         __ = 2  # [WPS121]
60 |         ___ = 3
61 |         _invalid_type = int  # [WPS121]
   |         ^^^^^^^^^^^^^ WPS121
62 |         isinstance(_, _invalid_type)
63 |         ___ = __ + _
   |
   = help: Remove leading underscores from variable `_invalid_type`

WPS121.py:67:9: WPS121 Local variable `_invalid_var_1` is marked as unused but is used
   |
66 |     def met(_valid_arg=3):
67 |         _invalid_var_1 = _valid_arg  # [WPS121]
   |         ^^^^^^^^^^^^^^ WPS121
68 |         
69 |         _invalid_var_2 = 1  # [WPS121]
   |
   = help: Remove leading underscores from variable `_invalid_var_1`

WPS121.py:69:9: WPS121 Local variable `_invalid_var_2` is marked as unused but is used
   |
67 |         _invalid_var_1 = _valid_arg  # [WPS121]
68 |         
69 |         _invalid_var_2 = 1  # [WPS121]
   |         ^^^^^^^^^^^^^^ WPS121
70 |         if _valid_arg:
71 |             return _invalid_var_1
   |
   = help: Remove leading underscores from variable `_invalid_var_2`
